src/plugins/home/server/services/sample_data/sample_data_registry.test.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { MockedKeys } from '@kbn/utility-types/jest';
+import { CoreSetup } from '../../../../../core/server';
+
+import { CustomIntegrationsPluginSetup } from '../../../../custom_integrations/server';
+import { customIntegrationsMock } from '../../../../custom_integrations/server/mocks';
+import { SampleDataRegistry } from './sample_data_registry';
+import { usageCollectionPluginMock } from '../../../../usage_collection/server/mocks';
+import { UsageCollectionSetup } from '../../../../usage_collection/server/plugin';
+import { coreMock } from '../../../../../core/server/mocks';
+
+describe('SampleDataRegistry', () => {
+  let mockCoreSetup: MockedKeys<CoreSetup>;
+  let mockCustomIntegrationsPluginSetup: MockedKeys<CustomIntegrationsPluginSetup>;
+  let mockUsageCollectionPluginSetup: MockedKeys<UsageCollectionSetup>;
+
+  beforeEach(() => {
+    mockCoreSetup = coreMock.createSetup();
+    mockCustomIntegrationsPluginSetup = customIntegrationsMock.createSetup();
+    mockUsageCollectionPluginSetup = usageCollectionPluginMock.createSetupContract();
+  });
+
+  describe('setup', () => {
+    test('should register the three sample datasets', () => {
+      const initContext = coreMock.createPluginInitializerContext();
+      const plugin = new SampleDataRegistry(initContext);
+      plugin.setup(
+        mockCoreSetup,
+        mockUsageCollectionPluginSetup,
+        mockCustomIntegrationsPluginSetup
+      );
+
+      const ids: string[] =
+        // @ts-expect-error registerCustomIntegration is jest-mock function
```

Comment:
Not 100% sure as I haven't tested this locally, but you might be able to avoid this by adjusting the types in the custom_integration plugin mocks. Looks like [it is using `MockedKeys`](https://github.com/elastic/kibana/blob/4c84644bb6fad2bb460a0db79690faac46a99e2b/src/plugins/custom_integrations/server/mocks.ts#L13-L20), but I'm not certain that this is necessary. You might be able to do:
```ts
const createCustomIntegrationsSetup = () =>  {
  const mock: jest.Mocked<CustomIntegrationsPluginSetup> = {
    registerCustomIntegration: jest.fn(),
    getAppendCustomIntegrations: jest.fn(),
  };

  return mock;
};
```

Reply:
I ran a type-check locally on the home-plugin, and seemed fine, see what CI will do with it 🤞